MaxText/layers/linears.py

```diff
@@ -382,17 +490,47 @@ def deepseek_scale_weights(self, weights):
     weights *= self.config.routed_scaling_factor
     return weights
 
-  def permute(self, inputs, gate_logits):
+  def deepseek_routing(self, gate_logits, pre_bias_logits):
```

Comment:
Can this be disabled, if users want to run with different routing?

Reply:
Add a TODO `TODO(ranran): Add an option to turn off device-limited routing